chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/assaults/ChaosMonkeyLatencyAssaultExecutor.java

```diff
@@ -0,0 +1,6 @@
+package de.codecentric.spring.boot.chaos.monkey.assaults;
+
+public interface ChaosMonkeyLatencyAssaultExecutor {
```

Comment:
I think the introduction of an interface for assaults makes sense. Should we rename this to `ChaosMonkeyAssaultExecutor`? After all I guess an interface makes sense if it is not bound to a specific assault right? Should we then start refactoring all assaults to implement that interface for their specific `execute`s? I am not that firm with the codebase so far, but to me it looks like this shouldnt be a specific assault interface.

Reply:
Yes, thats a good idea. In the beginning I had a similar idea, but thought it could be confused with the Interface ChaosMonkeyAssault. 

As far as I understand all Aussaults are split between ChaosMonkeyRequestAssault and ChaosMonkeyRuntimeAssault. Where as the Interface ChaosMonkeyAssault is a legacy one (and shouldnt be used anymore, according to javadoc)

The question remain, what will be the purpose of the ChaosMonkeyRequestAssault/ChaosMonkeyRuntimeAssault if we move the attack logic into separate executors? Are they just for metrics, stuff and the call of the executor?

Reply:
@MrBW can you give us a hint, whether the naming is fine like it is or whether going for a general `ChaosMonkeyAssaultExecutor` makes sense. In that case I guess we would gradually refactor and remove the other interfaces like `ChaosMonkeyRequestAssault`. Or maybe a short explanation why they are there makes sense, I guess I just don't get the idea so far.